### ISSUE TITLE
No fish_config file

### ISSUE THREAD

**Issue by nerd7473** [fish-shell/fish-shell#4287]
When you type the command fish_config you get the response:  ___** (exo-open:2315): WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-i1Wsz0QEA: Ceonnection refused

The issue opens yaourt-gui in a separate terminal window... I'm perplexed of this issue.

**Comment by krader1961** [comment]
This isn't a fish problem per se as fish does not send dbus messages. It's probably coming from your web browser. Also, you haven't provided enough information for us to even begin to help. OS? Fish version? Your default browser? Are you attempting to run `fish_config` from a ssh session or a native terminal?

Note that fish uses the Python `webbrowser` module to launch your web browser. We don't currently have any way to report which web browser it will attempt to run. But it's very likely it's trying to run whatever your default browser is as if you had launched it from the shell prompt.

**Comment by nerd7473** [comment]
I had run fish_config within terminal. Vivaldi is my browser. I'll check and make sure that is my default browser.

**Comment by nerd7473** [comment]
Changed the default browser and I have had the same issue. I changed it to Chromium this time. But the issue still persists, and the distribution which I so ignorantly forgot is Arch Labs... I have fish shell running in the same distro on my desktop PC... The desktop is my gaming rig. The laptop is the machine having said issue, first, should I remove the package and attempt a fresh install of fish shell. The package is in the community repo... Fish 2.6.0-1 I installed it with yaourt-gui would you recommend an uninstall of fish and a reinstall of it via  Pacman?

**Comment by nerd7473** [comment]
If any specific information is needed, I'll be happy to help provide it as to help the resolution of said issue.

**Comment by zanchey** [comment]
Does running `python3 -m "webbrowser" http://google.com` open Google in the right browser for you?

**Comment by krader1961** [comment]
Note that you have to actually provide a URL so @zanchey's command won't work as is. Try `python3 -m "webbrowser" -n http://fishshell.com`. But since that is what `fish_config` does I expect it to produce the same error. 

Basically, the "Couldn't connect to accessibility bus" error means your browser couldn't connect to the dbus service. This is not a fish issue. Reinstalling fish or doing anything to your fish config is not going to fix this. You've screwed up something on your laptop. Possibly the kernel is misconfigured. Possibly a permissions problem. Perhaps the `DBUS_SESSION_BUS_ADDRESS` variable is set incorrectly. Perhaps your `DISPLAY` variable is set incorrectly. You probably have a dbus daemon running but double-check by running `ps waux | grep dbus`.

Googling "Couldn't connect to accessibility bus: Failed to connect to socket" results in 58,600 results for me. I looked at a few of the results and saw quite a few different solutions.

For more information about D-Bus go here: https://www.freedesktop.org/wiki/Software/dbus/

**Comment by zanchey** [comment]
Er, I did provide a URL, but anyhow.

I think D-Bus is a furphy - my guess is that `exo-open` has a different idea of what the default browser is to the rest of your session.

**Comment by krader1961** [comment]
> Er, I did provide a URL, but anyhow.

Yes, you did. No idea why I didn't see that when I initially read your comment. Apologies.

Nonetheless, this isn't a fish issue. If you do figure out what is wrong with your laptop config, @nerd7473, please update this issue since the information may help someone else in the future.

**Comment by nerd7473** [comment]
@zanchey I'll look into exo-open

**Comment by nerd7473** [comment]
@krader1961 I'll try to see what the hell I did wrong...

**Comment by nerd7473** [comment]
@zanchey it opened in Vivaldi

**Comment by krader1961** [comment]
@nerd7473, I don't know what you mean when you say "it opened in Vivaldi". Are you saying that `fish_config` now works for you? If yes what did you change in your environment?

**Comment by faho** [comment]
@nerd7473: Ping?

Please clarify: Have you gotten it to work?

Or did you mean that `python3 -m "webbrowser" -n http://fishshell.com` opens in vivaldi, while `fish_config` does not?

If so, can you try `python3 -m "webbrowser" -n file:///some/path/to/some/file.html` (where that ".html" file actually exists?

**Comment by faho** [comment]
No answer in quite a while, closing.
